modules/dnn/test/test_onnx_importer.cpp

```diff
@@ -109,6 +109,7 @@ TEST_P(Test_ONNX_layers, MaxPooling_2)
 TEST_P(Test_ONNX_layers, Convolution)
 {
     testONNXModels("convolution");
+    testONNXModels("conv_asymmetric_pads");
```

Comment:
Is it possible to provide ONNX test data generator code ([`generate_onnx_models.py`](https://github.com/opencv/opencv_extra/blob/master/testdata/dnn/onnx/generate_onnx_models.py) in opencv_extra PR) ?

Reply:
No, I don't think its possible. torch.onnx.export() does not fuses padding with conv layer which is required to test this change.
As mentioned in description, tools like tf2onnx and keras2onnx use this type of fusion (onnx/tensorflow-onnx#334). The onnx model in opencv_extra has been generated using tf2onnx.

Reply:
> The onnx model in opencv_extra has been generated using tf2onnx.

tf2onnx has python interface, would it be possible to use that?

Reply:
> tf2onnx has python interface

Yes it can be used, but the original network is built using tensorflow. `generate_onnx_models.py` uses Pytorch.
Code used to generate onnx model
```python3
import tensorflow as tf
import tf2onnx
tf.keras.backend.set_image_data_format('channels_first')

inp = tf.keras.layers.Input(batch_shape = (1, 2, 3, 4))
pad = tf.keras.layers.ZeroPadding2D(padding = ((2, 1), (2, 1)))(inp)
out = tf.keras.layers.Conv2D(3, kernel_size = 3)(pad)
model = tf.keras.Model(inp, out)
onnx_model, _ = tf2onnx.convert.from_keras(model, output_path="/conv_asymmetric_pads.onnx")
```

Reply:
Perhaps it makes sense to create new file like `generate_onnx_testdata_tf.py` (or tf2 to mention TensorFlow 2.x).

BTW, Old file can be renamed to `generate_onnx_testdata_pytorch.py` separately.

Reply:
Sounds good. Should I add these 2 changes in opencv_extra PR?

Reply:
Feel free to create a new file in opencv_extra.
It would be nice if you can prepare `requirements-tf.txt` file too (list required pip packages there, `pip freeze` may help with that, [example](https://github.com/opencv/opencv/blob/master/samples/dnn/dnn_model_runner/dnn_conversion/requirements.txt))